wandb/sdk/interface/artifacts.py

```diff
@@ -805,72 +837,76 @@ def store_path(
-        self._artifacts_by_id = {}
+        self._artifacts_by_id: Dict[str, Artifact] = {}
         self._random = random.Random()
         self._random.seed()
-        self._artifacts_by_client_id = {}
+        self._artifacts_by_client_id: Dict[str, "wandb_artifacts.Artifact"] = {}
 
     def check_md5_obj_path(
         self, b64_md5: B64MD5, size: int
-    ) -> Tuple[util.FilePathStr, bool, "Opener"]:
+    ) -> Tuple[FilePathStr, bool, "Opener"]:
         hex_md5 = b64_to_hex_id(b64_md5)
         path = os.path.join(self._cache_dir, "obj", "md5", hex_md5[:2], hex_md5[2:])
         opener = self._cache_opener(path)
         if os.path.isfile(path) and os.path.getsize(path) == size:
-            return path, True, opener
-        filesystem.mkdir_exists_ok(os.path.dirname(path))
-        return path, False, opener
+            return FilePathStr(path), True, opener
+        mkdir_exists_ok(os.path.dirname(path))
+        return FilePathStr(path), False, opener
 
     # TODO(spencerpearson): this method at least needs its signature changed.
     # An ETag is not (necessarily) a checksum.
     def check_etag_obj_path(
         self,
-        url: util.URIStr,
+        url: URIStr,
         etag: ETag,
         size: int,
-    ) -> Tuple[util.FilePathStr, bool, "Opener"]:
+    ) -> Tuple[FilePathStr, bool, "Opener"]:
         hexhash = hashlib.sha256(
             hashlib.sha256(url.encode("utf-8")).digest()
             + hashlib.sha256(etag.encode("utf-8")).digest()
         ).hexdigest()
         path = os.path.join(self._cache_dir, "obj", "etag", hexhash[:2], hexhash[2:])
         opener = self._cache_opener(path)
         if os.path.isfile(path) and os.path.getsize(path) == size:
-            return path, True, opener
-        filesystem.mkdir_exists_ok(os.path.dirname(path))
-        return path, False, opener
+            return FilePathStr(path), True, opener
+        mkdir_exists_ok(os.path.dirname(path))
+        return FilePathStr(path), False, opener
 
-    def get_artifact(self, artifact_id):
+    def get_artifact(self, artifact_id: str) -> Optional["Artifact"]:
         return self._artifacts_by_id.get(artifact_id)
 
-    def store_artifact(self, artifact):
+    def store_artifact(self, artifact: "Artifact") -> None:
+        if not artifact.id:
+            raise ArtifactNotLoggedError(artifact, "store_artifact")
         self._artifacts_by_id[artifact.id] = artifact
 
-    def get_client_artifact(self, client_id):
+    def get_client_artifact(
+        self, client_id: str
+    ) -> Optional["wandb_artifacts.Artifact"]:
         return self._artifacts_by_client_id.get(client_id)
 
-    def store_client_artifact(self, artifact):
+    def store_client_artifact(self, artifact: "wandb_artifacts.Artifact") -> None:
         self._artifacts_by_client_id[artifact._client_id] = artifact
 
     def cleanup(self, target_size: int) -> int:
-        bytes_reclaimed: int = 0
-        paths: Dict[os.PathLike, os.stat_result] = {}
```

Comment:
Oh, this one was wrong because what gets returned is a `str` and so not actually a `os.PathLike`, but mypy can infer that from the later assignments and at least in this case with this version of mypy it doesn't ask for a type declaration.